tracing-subscriber/src/layer.rs

```diff
@@ -1114,15 +1113,20 @@ where
     /// [stored data]: ../registry/struct.SpanRef.html
     #[cfg(feature = "registry")]
     #[cfg_attr(docsrs, doc(cfg(feature = "registry")))]
+    #[deprecated(
+        note = "equivalent to self.lookup_current().into_iter().flat_map(|span| span.scope().from_root()), but consider whether lookup_current is a bug"
```

Comment:
hmm. i'm not a huge fan of deprecating this without adding an alternative. it feels like the suggested replacement involves significantly more boilerplate than this method, and i think it's good to have the `Context` type provide a way to iterate the current scope easily --- that seems like a fairly common thing to want to do. 

also, if we _are_ going to deprecate this, i'm not a big fan of deprecating it with the message "but consider whether this is a bug", without any explanation of _why_ it might be a bug. to me, it feels like "but consider whether this is a bug" is just going to confuse and scare people.

IMO, what i'd rather do is rename this method to `current_scope`; explain clearly in the docs that this returns the scope of whatever span is _contextually current_; and add a `fn event_scope(&self, event: &Event<'_>) -> Scope<'_, S>` to `Context`, which would return either the scope of the event's overridden parent _or_ call `current_scope` if the event is contextual. that way, users would still have shorthand for asking the context what the current scope is, but the distinction between the current scope and the possibly overridden event scope would be clearer.

what do you think?

Reply:
> If you're writing a span handler then you already have an span ID, and you'd want to iterate from there (so roughly `ctx.span(id)?.scope()`).

hmm, this is a good point! i was thinking that this would be useful in all the span handlers, but you're right that in that case, the layer can look up the span based on the ID that's passed in.

i do think, though, that it would be nice to have helpers to avoid having to write out all the boilerplate to look up a span and iterate over its scope if it exists. i think `event_scope` and `span_scope` could be worth having for that reason. i think that it would be better to be able to suggest one to two method calls, rather than having to recommend a pretty big blob of code as a replacement.

Reply:
Alright, thinking about it a bit more, `span_scope` shouldn't be a problem, but `event_scope` kind of depends on #1434.

Reply:
Added `span_scope` now, waiting for #1434 before we can add `event_scope`